packages/magento-product/components/ProductFiltersPro/FilterLayout.tsx

```diff
@@ -0,0 +1,131 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { CategoryQueryFragment } from '@graphcommerce/magento-category/queries/CategoryQueryFragment.gql'
+import { StickyBelowHeader, responsiveVal } from '@graphcommerce/next-ui'
+import { Box, Container, SxProps, Theme } from '@mui/material'
+import React from 'react'
+import { ProductListQuery } from '../ProductList/ProductList.gql'
+import { ProductListCount as ProductListCountElement } from '../ProductListCount/ProductListCount'
+import { ProductFiltersQuery, ProductListFilters } from '../ProductListFilters'
+import { ProductListFiltersContainer } from '../ProductListFiltersContainer/ProductListFiltersContainer'
+import { ProductListItems as ProductListItemsElement } from '../ProductListItems/ProductListItems'
+import { ProductListParamsProvider } from '../ProductListItems/ProductListParamsProvider'
+import { FilterTypes, ProductListParams } from '../ProductListItems/filterTypes'
+import { ProductListPagination as ProductListPaginationElement } from '../ProductListPagination/ProductListPagination'
+import { ProductListSort } from '../ProductListSort'
+import { ProductFiltersPro } from './ProductFiltersPro'
+import { ProductFiltersProAllFiltersChip } from './ProductFiltersProAllFiltersChip'
+import { ProductFiltersProAllFiltersSidebar } from './ProductFiltersProAllFiltersSidebar'
+import { ProductFiltersProFilterChips } from './ProductFiltersProChips'
+import { ProductFiltersProLimitChip } from './ProductFiltersProLimitChip'
+import { ProductFiltersProSortChip } from './ProductFiltersProSortChip'
+
+export interface FilterLayoutProps
+  extends NonNullable<ProductListQuery>,
+    NonNullable<ProductFiltersQuery> {
+  ProductListItems: React.FC<React.ComponentProps<typeof ProductListItemsElement>>
+  ProductListCount: React.FC<React.ComponentProps<typeof ProductListCountElement>>
+  ProductListPagination: React.FC<React.ComponentProps<typeof ProductListPaginationElement>>
+  filterTypes: FilterTypes
+  params: ProductListParams
+  category: NonNullable<NonNullable<CategoryQueryFragment['categories']>['items']>[number]
+  mode?: 'default' | 'sidebar'
+  sx?: SxProps<Theme>
+}
+
+export function FilterLayout(props: FilterLayoutProps) {
+  const {
+    ProductListItems,
+    ProductListCount,
+    ProductListPagination,
+    products,
+    filterTypes,
+    filters,
+    params,
+    category,
+    mode = 'default',
+    sx = [],
+  } = props
+
+  if (mode === 'sidebar' && import.meta.graphCommerce.productFiltersPro) {
+    return (
+      <Container maxWidth='lg' sx={Array.isArray(sx) ? sx : [sx]}>
+        <Box
+          sx={(theme) => ({
+            display: 'grid',
+            gridTemplateColumns: '3fr 9fr',
+            columnGap: theme.spacings.md,
+            [theme.breakpoints.down('md')]: { display: 'flex' },
```

Comment:
- Use responsive object values whenever possible: `display: { xs: 'flex', md: 'flex'}`
- Please develop mobile first from a code perspective, don't use `down`, use `up('md')` whenever possible.
- Do not switch rendering technique for mobile / desktop. Choose grid or flex for both